### PR TITLE
Unit Tests for Node Port Local

### PR DESCRIPTION
**Pull request by monotosh-avi**
- Added UTs for NPL
- Fixed issues in the code caught by UTs

### CODE REVIEW

**Comment by anish-avi** [comment]
build ako

**Comment by anish-avi** [comment]
build ako

**Comment by anish-avi** [comment]
build ako

**Comment by anish-avi** [comment]
build ako

**Comment by anish-avi** [comment]
build ako

**Comment by anish-avi** [comment]
build ako

**Comment by anish-avi** [comment]
Testing auto-comment

**Comment by anish-avi** [comment]
build ako

**Comment by sudswasavi** [comment]
Hi @shubhamavi / @monotosh-avi  -   oshift_crd_test.go:244: this test is flakey, I increased the timer to 90 seconds but I still it's failing intermittently. Can you PTAL and see what is going on?

**Comment by shubhamavi** [comment]
build ako

**Review by sudswasavi — COMMENTED** [review]
@monotosh-avi good coverage in the tests. What happens if you had selected NPL as the serviceType and someone updated it to ClusterIP and restarted AKO?

**Comment by monotosh-avi** [comment]
build ako

**Comment by monotosh-avi** [comment]
> @monotosh-avi good coverage in the tests. What happens if you had selected NPL as the serviceType and someone updated it to ClusterIP and restarted AKO?

Then the old server IPs would be removed and new servers would be added as per the PodIP. @sudswasavi

File: tests/npltests/npl_pod_test.go [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+
+	var aviModel interface{}
+	var found bool
+	g.Eventually(func() bool {
+		found, aviModel = objects.SharedAviGraphLister().Get(defaultLBModel)
+		return aviModel == nil
+	}, 40*time.Second).Should(gomega.Equal(false))
+	if !found {
+		t.Fatalf("Couldn't find model %v", integrationtest.SINGLEPORTMODEL)
+	} else {
```
Comment (parimanur) [inline tests/npltests/npl_pod_test.go:471]: Else is not required.

Comment (monotosh-avi) [reply tests/npltests/npl_pod_test.go]: Good catch. Updated.

**Review by shubhamavi — APPROVED** [review]

**Review by hemantavi — APPROVED** [review]

**Pull request merged**